elastic/apm-server
# modelindexer: Fix TooManyRequests stat reporting

## Motivation/summary

Fixes a bug where the TooManyRequests (429) response from Elasticsearch
is not correctly reported, since it's returned as an error.

## Checklist



~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/master/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

* modify `docker-compose.yml` so that the Elasticsearch container has 4Gb of memory assigned to it.
* apply this patch:
```patch
diff --git a/model/modelindexer/indexer.go b/model/modelindexer/indexer.go
index f1ba63815..f12100f0c 100644
--- a/model/modelindexer/indexer.go
+++ b/model/modelindexer/indexer.go
@@ -129,7 +129,7 @@ func New(client elasticsearch.Client, cfg Config) (*Indexer, error) {
 		)
 	}
 	if cfg.MaxRequests <= 0 {
-		cfg.MaxRequests = 10
+		cfg.MaxRequests = 15
 	}
 	if cfg.FlushBytes <= 0 {
 		cfg.FlushBytes = 5 * 1024 * 1024
```
* `./apm-server -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme -E apm-server.expvar.enabled=true`
* Run this load generator: https://github.com/elastic/apm-server/blob/931cdc43f6a1766fbfed7cf07c9c66cc97dfe1bf/testing/load_generator/main.go. You may need to increase the concurrency numbers to see the stats being reported.
* Navigate to http://localhost:8200/debug/vars and check `libbeat.output.events.toomany` is eventually > 0
